app/services/notifications/notifiable_action/send.rb

```diff
@@ -16,16 +16,16 @@ def self.call(...)
       end
 
       def call
+        return unless notifiable.is_a?(Article)
+
         json_data = {
           user: user_data(notifiable.user),
           article: article_data(notifiable)
         }
         json_data[:organization] = organization_data(notifiable.organization) if notifiable.organization_id
 
         notifications_attributes = []
-        # followers is an array and not an activerecord object
-        # followers can occasionally be nil because orphaned follows can possibly exist in the db (for now)
-        followers.sort_by(&:updated_at).reverse[0..10_000].each do |follower|
+        notifiable.followers.sort_by(&:updated_at).reverse[0..10_000].each do |follower|
```

Comment:
**suggestion (non-blocking):** If this array has more than 10k elements we first reverse all of them and then throw some away. The following avoids this by using `last` and [`reverse_each`](https://docs.ruby-lang.org/en/2.7.0/Array.html#method-i-reverse_each) which generally should also be faster than calling `reverse.each`.

```suggestion
        notifiable.followers.sort_by(&:updated_at).last(10_000).reverse_each do |follower|
```

Reply:
Oh, good catch! I also never get the chance to use `reverse_each`, so this is a double win 😃